src/Services/AttributesService.php

```diff
@@ -131,33 +131,36 @@ private function getFieldDefinitionsRecursive($class, &$attributes, $prefix, arr
                 $allowedTypes = $field->getAllowedTypes();
                 foreach ($allowedTypes as $allowedType) {
                     $allowedTypeClass = ObjectbrickDefinition::getByKey($allowedType);
-                    $this->getFieldDefinitionsRecursive($allowedTypeClass, $attributes, $prefix . $field->getName() . "." . $allowedType . ".", $checkedClasses);
+                    $this->getFieldDefinitionsRecursive($allowedTypeClass, $attributes, $prefix . $field->getName() . "." . $allowedType . "." . ($suffix ? $suffix . "." : ""), $checkedClasses);
                 }
             } elseif ($field instanceof Fieldcollections) {
                 $allowedTypes = $field->getAllowedTypes();
                 foreach ($allowedTypes as $allowedType) {
                     $allowedTypeClass = FieldcollectionDefinition::getByKey($allowedType);
-                    $this->getFieldDefinitionsRecursive($allowedTypeClass, $attributes, $prefix . $field->getName() . ".", $checkedClasses);
+                    $this->getFieldDefinitionsRecursive($allowedTypeClass, $attributes, $prefix . $field->getName() . "." . ($suffix ? $suffix . "." : ""), $checkedClasses);
                 }
             } elseif ($field instanceof ClassificationStoreDefinition) {
                 $fields = $this->getStoreKeys($field->getStoreId());
                 foreach ($fields as $field) {
                     $attributes[] = $prefix . $field->getName();
                 }
             } elseif ($field instanceof Localizedfields) {
+                $langs = \Pimcore\Tool::getValidLanguages();
                 $fields = $field->getChildren();
                 foreach ($fields as $childField) {
                     if (!method_exists($childField, "getFieldDefinitions")) {
-                        $attributes[] = $prefix . $childField->getName();
+                        $attributes = array_merge($attributes, array_map(fn ($lang) => $prefix . $childField->getName() . "." . $lang, $langs));
                     } else {
-                        $this->getFieldDefinitionsRecursive($childField, $attributes, $prefix . $childField->getName() . ".", $checkedClasses);
+                        $this->getFieldDefinitionsRecursive($childField, $attributes, $prefix . $childField->getName() . "." . ($suffix ? $suffix . "." : ""), $checkedClasses);
```

Comment:
Pulling the third parameter out to a sensibly named local var should help readability